lib/RouterMixin.js

```diff
@@ -203,7 +203,12 @@ var RouterMixin = {
 };
 
 function join(a, b) {
-  return (a + b).replace(/\/\//g, '/');
+  if ((a+b).search(/^http/i) > -1) {
```

Comment:
Ah okay.

The `//` replace has to do with joining; basically it's a fast way of
making sure "app/" + "/foo" (contextual router prefix + route) doesn't
turn into "app//foo".
On 1/12/15 9:31 PM, Fede Torre Mtz wrote:
>
> In lib/RouterMixin.js
> <https://github.com/STRML/react-router-component/pull/112#discussion-diff-22819577>:
>
> > @@ -203,7 +203,12 @@ var RouterMixin = {
> >  };
> >  
> >  function join(a, b) {
> > -  return (a + b).replace(/\/\//g, '/');
> > +  if ((a+b).search(/^http/i) > -1) {
>
> This is called when the component is created. I'm not sure why // gets
> replaced with a single / -- but in the case of a full url link we do
> not want that behavior. Or else our element's would be
> http:/example.com instead of <%7Blink%7D>http://example.com
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/STRML/react-router-component/pull/112/files#r22819577>.
>